### PR TITLE
Login

### LINKED ISSUE #10: Log in views
## REQUIREMENTS / DESCRIPTION
make mobile mouck up for log in views

## DESIGN
image

### PR DESCRIPTION
**Pull request by Bryanroal**
Responsive login for mobile and desktop

Login Desktop
Login Movil

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "liveServer.settings.port": 5501
+}
```
Comment (MauricioNC) [inline .vscode/settings.json]: Add an empty line at the end of your code

File: app/assets/config/manifest.js [hunk 2]
```diff
@@ -2,3 +2,4 @@
 //= link_directory ../stylesheets .css
 //= link_tree ../../javascript .js
 //= link_tree ../../../vendor/javascript .js
+//= link main.css
```
Comment (MauricioNC) [inline app/assets/config/manifest.js]: Add an empty line at the end of your code

File: style.css [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+.register-link p a {
+  color: #000000;
+  text-decoration: none;
+  font-weight: 600;
+}
+
+.register-link p a:hover {
+  text-decoration: underline;
+}/*# sourceMappingURL=style.css.map */
```
Comment (MauricioNC) [inline style.css]: Add an empty line at the end of your code

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (juperameza) [reply .vscode/settings.json]: More like this shouldnt be here at all

File: app/assets/stylesheets/main.scss [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        border: none;
+        color: #333;
+        cursor: pointer;
+        font-size: 16px;
+        font-weight: 600;
+        height: 40px;
+        outline: none;
+        align-items: center;
+        width: 80%;
+        margin: 0 10px 20px 30px
```
Comment (juperameza) [inline app/assets/stylesheets/main.scss]: Order styles alphabetically

File: app/views/devise/shared/_links.html.erb [hunk 5]
```diff
@@ -3,7 +3,7 @@
 <% end %>
 
 <%- if devise_mapping.registerable? && controller_name != 'registrations' %>
-  <%= link_to "Sign up", new_registration_path(resource_name) %>
+  <%= link_to "Sign up" , new_registration_path(resource_name), class:"sig" %>
```
Comment (juperameza) [inline app/views/devise/shared/_links.html.erb]: You can name this class better, what is sig?

File: style.css [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  margin: 20px 0 15px;
+}
+
+.register-link p a {
+  color: #000000;
+  text-decoration: none;
+  font-weight: 600;
+}
+
+.register-link p a:hover {
```
Comment (juperameza) [inline style.css]: Why is this file here???????

**Review by juperameza — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/main.scss [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+    .register-link p a:hover {
+        text-decoration: underline;
+    }
+    
+    .sig{
+        margin: 0 40px 0 40px;
+    
+    }
+}
+
```
Comment (juperameza) [inline app/assets/stylesheets/main.scss]: I recommend using something like sass 7.1 pattern to order files since as your application grows it will get harder and harder to mantain it

**Review by juperameza — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/main.scss (hunk 4 above)
Comment (Bryanroal) [reply app/assets/stylesheets/main.scss]: okay

File: style.css (hunk 3 above)
Comment (Bryanroal) [reply style.css]: okay

**Review by MauricioNC — APPROVED** [review]

File: app/assets/config/manifest.js [hunk 8]
```diff
@@ -2,3 +2,4 @@
 //= link_directory ../stylesheets .css
 //= link_tree ../../javascript .js
 //= link_tree ../../../vendor/javascript .js
```
Comment (juperameza) [inline app/assets/config/manifest.js:4]: You didn't add the an empty line

File: .vscode/settings.json (hunk 1 above)
Comment (juperameza) [inline .vscode/settings.json]: Why is this file here???

File: app/assets/stylesheets/abstracts/_mixins.scss:Zone.Identifier [hunk 9]
```diff
@@ -0,0 +1,3 @@
+[ZoneTransfer]
+ZoneId=3
+HostUrl=https://files.slack.com/files-pri/TGJT2DH3L-F06NQBPEWHG/download/_mixins.scss?origin_team=TGJT2DH3L
```
Comment (juperameza) [inline app/assets/stylesheets/abstracts/_mixins.scss:Zone.Identifier]: what is this??

File: app/assets/stylesheets/components/_form.scss [hunk 10]
```diff
@@ -0,0 +1,158 @@
+*{
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+
+}
```
Comment (juperameza) [inline app/assets/stylesheets/components/_form.scss]: I wouldn't recoommmend changing everything here with the form

File: style.css (hunk 3 above)
Comment (juperameza) [inline style.css]: new line at the end of this file!!!

**Review by juperameza — CHANGES_REQUESTED** [review]

**Review by juperameza — CHANGES_REQUESTED** [review]
Also resolver ocnflicts!!!!!!!

File: app/assets/stylesheets/components/_buttons.scss [hunk 11]
```diff
@@ -0,0 +1,69 @@
+.btn{
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_buttons.scss]: add a space before the curly brace

File: app/assets/stylesheets/components/_buttons.scss [hunk 12]
```diff
@@ -0,0 +1,69 @@
+.btn{
+    
+    align-items: center;
+    background: #fff;
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_buttons.scss]: Add this color in `color.scss` file and call it like `$white`

File: app/assets/stylesheets/components/_buttons.scss [hunk 13]
```diff
@@ -0,0 +1,69 @@
+.btn{
+    
+    align-items: center;
+    background: #fff;
+    border-radius: 40px;
+    border: none;
+    color: #333;
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_buttons.scss]: same here

File: app/assets/stylesheets/components/_buttons.scss [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+        font-size: 16px;
+        font-weight: 600;
+        height: 40px;
+        margin: 0 10px 20px 30px;
+        outline: none;
+        width: 80%;
+    }
+    .center{
+        align-content: center;
+    }
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_buttons.scss]: Fix the indentation at 2 spaces and add spaces before the curly braces in this piece of code and all this file, because is fundamental to set a legible format.

File: app/assets/stylesheets/components/_buttons.scss [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+        font-weight: 600;
+
+        outline: none;
+        height: 40px;
+        width: 80%; 
+        }
+    .center{
+        align-content: center;
+    }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_buttons.scss]: 😳

File: app/assets/stylesheets/components/_form.scss [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+
+}
+
+
+.wrapper {
+
+    background: #D9D9D9;
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_form.scss]: same here

File: app/assets/stylesheets/components/_form.scss [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+    border-radius: 30px;
+    color: #000000;
+    margin: auto;
+    width: 75%;
+    height: 55%;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_form.scss]: sort alphabetically and fix the indentation

File: app/assets/stylesheets/components/_form.scss [hunk 18]
```diff
@@ -0,0 +1,158 @@
...
+    
+    .register-link p a:hover {
+        text-decoration: underline;
+    }
+    
+    .signuplabel{
+        margin: 0 40px 0 40px;
+    
+    }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_form.scss]: Rewrite all this code with an improved format.

File: app/assets/stylesheets/components/_input.scss [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+        outline: none;
+        padding: 20px 45px 20px 20px;
+        width: 100%;
+    }
+    
+    .input-box input::placeholder{
+        color: #000000;
+    }
+
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_input.scss]: same here

File: app/assets/stylesheets/main.scss [hunk 20]
```diff
@@ -1,23 +1,33 @@
...
-  'components/posts';
-
-@import
-  'pages/feed';
+@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600;700;800;900&display=swap");
+@import url("https://fonts.googleapis.com/css?family=Black+Ops+One");
+@import "abstracts/mixins";
+
+
+
```
Comment (JuanequeX) [inline app/assets/stylesheets/main.scss]: delete these extra spaces

File: app/views/devise/sessions/new.html.erb [hunk 21]
```diff
@@ -1,27 +1,33 @@
-<h2>Log in</h2>
 
+<div class="logo">
+        <%= image_tag("gymratlogo.png") %>
+        <h1>GYMRAT</h1>
+    </div>
```
Comment (JuanequeX) [inline app/views/devise/sessions/new.html.erb]: fix this indentation

File: app/views/devise/sessions/new.html.erb [hunk 22]
```diff
@@ -1,27 +1,33 @@
...
+<div class="wrapper">
+<h2>Log in</h2>
 <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
-  <div class="field">
+  <div class="input-box">
     <%= f.label :email %>
-    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+    <%= f.email_field :email, autofocus: true, autocomplete: "email", class:"input-field" %>
```
Comment (JuanequeX) [inline app/views/devise/sessions/new.html.erb]: Add a `required: true` for join validation on front

Comment (JuanequeX) [reply app/views/devise/sessions/new.html.erb]: in each input

File: app/views/layouts/application.html.erb [hunk 23]
```diff
@@ -6,6 +6,7 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+
```
Comment (JuanequeX) [inline app/views/layouts/application.html.erb]: ???

File: style.css [hunk 24]
```diff
@@ -0,0 +1,124 @@
...
+  width: 100%;
+  height: 45px;
+  background: #fff;
+  border: none;
+  outline: none;
+  border-radius: 40px;
+  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+  cursor: pointer;
+  font-size: 16px;
+  color: #333;
```
Comment (JuanequeX) [inline style.css]: sort this block

File: style.css [hunk 25]
```diff
@@ -0,0 +1,124 @@
...
+  width: 100%;
+  height: 100%;
+  background: transparent;
+  border: none;
+  outline: none;
+  border: 2px solid rgba(255, 255, 255, 0.2);
+  border-radius: 40px;
+  font-size: 16px;
+  color: #0a0a0a;
+  padding: 20px 45px 20px 20px;
```
Comment (JuanequeX) [inline style.css]: sort this block

File: style.css [hunk 26]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+body {
+  gap: 1.5rem;
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  min-height: 100vh;
+  background: #FEFCE8;
```
Comment (JuanequeX) [inline style.css]: sort this block

**Review by JuanequeX — CHANGES_REQUESTED** [review]
I believe that it's a good idea to centralize code styles, whether it's in CSS or SCSS.

File: style.css [hunk 27]
```diff
@@ -0,0 +1,124 @@
+@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600;700;800;900&display=swap");
+@import url("https://fonts.googleapis.com/css?family=Black+Ops+One");
```
Comment (juperameza) [inline style.css]: Lol why is this file still here ??

**Review by juperameza — CHANGES_REQUESTED** [review]

File: style.css (hunk 27 above)
Comment (JuanequeX) [reply style.css]: dimelo tu

**Review by Tole15 — APPROVED** [review]

File: app/views/devise/sessions/new.html.erb (hunk 21 above)
Comment (Bryanroal) [reply app/views/devise/sessions/new.html.erb]: okay

File: app/assets/stylesheets/main.scss (hunk 20 above)
Comment (Bryanroal) [reply app/assets/stylesheets/main.scss]: okay

File: app/assets/stylesheets/components/_buttons.scss (hunk 12 above)
Comment (Bryanroal) [reply app/assets/stylesheets/components/_buttons.scss]: okay

File: app/assets/stylesheets/components/_buttons.scss (hunk 14 above)
Comment (Bryanroal) [reply app/assets/stylesheets/components/_buttons.scss]: okay

File: style.css (hunk 24 above)
Comment (Bryanroal) [reply style.css]: okay

**Review by JuanequeX — APPROVED** [review]

File: app/assets/stylesheets/components/_input.scss
Comment (angel23v) [inline app/assets/stylesheets/components/_input.scss:1]: Decide how you are gonna set your curly braces

class{

}

or

class
   }

   }

**Review by angel23v — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/components/_input.scss [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+        padding: 20px 45px 20px 20px;
+        width: 100%;
+    }
+    
+    .input-box input::placeholder
+    {
+        color:$color-light;
+    }
+
+}
```
Comment (juperameza) [inline app/assets/stylesheets/components/_input.scss]: New line at the end of the code

File: img/gymratlogo.png:Zone.Identifier [hunk 29]
```diff
@@ -0,0 +1,3 @@
+[ZoneTransfer]
+ZoneId=3
+HostUrl=https://www.figma.com/
```
Comment (juperameza) [inline img/gymratlogo.png:Zone.Identifier:3]: What is this?

**Review by juperameza — CHANGES_REQUESTED** [review]

**Review by angel23v — APPROVED** [review]

**Review by juperameza — APPROVED** [review]
